src/main/java/org/infinispan/persistence/mongodb/config/ConnectionConfiguration.java

```diff
@@ -0,0 +1,45 @@
+package org.infinispan.persistence.mongodb.config;
+
+import org.infinispan.commons.configuration.attributes.Attribute;
+import org.infinispan.commons.configuration.attributes.AttributeDefinition;
+import org.infinispan.commons.configuration.attributes.AttributeSet;
+import org.infinispan.commons.configuration.attributes.ConfigurationElement;
+import org.infinispan.configuration.cache.AbstractStoreConfiguration;
+
+
+public class ConnectionConfiguration extends ConfigurationElement<ConnectionConfiguration> {
+
+    static final AttributeDefinition<String> URI = AttributeDefinition.builder(org.infinispan.persistence.mongodb.config.Attribute.CONNECTION_URI, null, String.class).immutable().build();
+    static final AttributeDefinition<String> DATABASE = AttributeDefinition.builder(org.infinispan.persistence.mongodb.config.Attribute.DATABASE, null, String.class).immutable().build();
+    static final AttributeDefinition<String> COLLECTION = AttributeDefinition.builder(org.infinispan.persistence.mongodb.config.Attribute.COLLECTION, null, String.class).immutable().build();
+
+    public static AttributeSet attributeDefinitionSet() {
+        return new AttributeSet(ConnectionConfiguration.class, AbstractStoreConfiguration.attributeDefinitionSet(),
+                URI, DATABASE, COLLECTION
+        );
+    }
+
+
+    private final Attribute<String> uri;
```

Comment:
no need to have these attributes as fields, since they will be accessed infrequently. Just reference the attributes in the getters below